link/platform.go

```diff
@@ -0,0 +1,19 @@
+package link
+
+import (
+	"fmt"
+	"runtime"
+)
+
+func platformPrefix(symbol string) string {
+
+	prefix := runtime.GOARCH
+	switch prefix {
```

Comment:
Do you have a reference / link which prefixes the kernel uses? Your code currently assumes that GOARCH == kernel prefix in general, I'm not sure that will be true.

Reply:
> Your code currently assumes that GOARCH == kernel prefix in general, I'm not sure that will be true.

Out of curiosity, what do you mean by this? I fail to come up with an example. Perhaps when running the binary under `qemu-user-static`? What other facilities would the runtime expose to detect underlying OS?

Reply:
> Do you have a reference / link which prefixes the kernel uses?

Tricky, but from what I can find:

```
# in linux
λ  linux master ✗ egrep -R "define __SYSCALL(.*) = __" arch
arch/x86/entry/syscall_64.c:#define __SYSCALL_64(nr, sym) [nr] = __x64_##sym,
arch/x86/entry/syscall_32.c:#define __SYSCALL_I386(nr, sym) [nr] = __ia32_##sym,
arch/x86/entry/syscall_x32.c:#define __SYSCALL_X32(nr, sym) [nr] = __x32_##sym,
arch/x86/entry/syscall_x32.c:#define __SYSCALL_COMMON(nr, sym) [nr] = __x64_##sym,
arch/arm64/kernel/sys32.c:#define __SYSCALL(nr, sym)	[nr] = __arm64_##sym,
arch/arm64/kernel/sys.c:#define __SYSCALL(nr, sym)	[nr] = __arm64_##sym,
```

Reply:
> Out of curiosity, what do you mean by this?

The code relies on the fact that on arm64 GOARCH == the prefix used by the kernel, and assumes that the same is true for other platforms. Nothing to do with changes in GOARCH due to different runtimes.

Reply:
Oh, completely missed that. We should probably return empty by default and only emit known prefixes.

@iii-i (sorry for the random ping), could you chime in on what's common in s390(x)? I'm finding references to both `__s390_` and `__s390x_` in Linux.

Reply:
I assume we can cover as much as possible of the options. I tried to cover x86 and arm64. We can panic on unknown platform and let these be added over time once users hit "unsupported platform".

Reply:
As this is library code, we cannot panic. The safer option would be to not return a prefix, so the given symbol is plumbed through as is. Of course, we'd want to ship this with most of the available architectures supported as you suggested..